### PR TITLE
MDLSITE-6745 mustachelint: Ensure that fatal errors include a source

### PR DESCRIPTION
**Pull request by andrewnicols**
From: https://tracker.moodle.org/browse/MDLSITE-6745

### CODE REVIEW

**Comment by stronk7** [comment]
I've been reading the Tracker issue and it's not clear for me if this should be considered ready or no. Please, confirm.

No matter of that, a couple of comments about the current patch could be:

- Do we need to amend any bats test (or create a new one) to ensure the new information is correctly being output?
- It seems that this needs a rebase on top of current master.

Ciao :-)

**Comment by andrewnicols** [comment]
> Do we need to amend any bats test (or create a new one) to ensure the new information is correctly being output?

I've added a new test for this.

> It seems that this needs a rebase on top of current master.

Done. Only one commit missing /o\

File: tests/fixtures/31-mustache_lint-js_token_test.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
+From 7c4ad58c7f7a462e39577959a6c4e05a868cc610 Mon Sep 17 00:00:00 2001
+From: Andrew Nicols <andrew@nicols.co.uk>
+Date: Mon, 8 Aug 2022 15:07:36 +0800
+Subject: [PATCH 1/1] MDL-12345 mustsache: fixture for js parsing failure
```
Comment (scara) [inline tests/fixtures/31-mustache_lint-js_token_test.patch:4]: Small typo? `mustsache` => `mustache`

HTH,
Matteo

Comment (stronk7) [reply tests/fixtures/31-mustache_lint-js_token_test.patch:4]: You are right, @scara . Though I must confess how "mustsache" sounded in my brain, LOL.

Comment (stronk7) [reply tests/fixtures/31-mustache_lint-js_token_test.patch]: I've taken the liberty of amending the typo adding extra commit, will squash on merge.

**Comment by stronk7** [comment]
(have squashed the 2 commits, so it's all together in one now)

**Comment by stronk7** [comment]
Here we go, used "rebase + merge" coz there was another patch landed mins ago.

Thanks all, ciao :-)

**Pull request merged**